### PR TITLE
Grd issues 401

### PR DESCRIPTION
**Pull request by raftaar1191**
## Description
PR for issues https://github.com/WordImpress/Give-Recurring-Donations/issues/401

`give_delete_donation` of Give Core functions handle all the related thing that we need to do 
1>Like deleting a log
2> Updating User
3> Deleting MetaData for the Donations
4> Delete COmment Related to Donations 
5> Plus have some predefine hooks to handle the thing as well 

Here we can see that we are directly firing the SQL which is not good as we already have a specific function to delete the Donations. So Using it is a good options

## How Has This Been Tested?
Manually test By deleting All the test Donations one by one
Note: Please also merge https://github.com/WordImpress/Give-Recurring-Donations/pull/407

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/admin/tools/data/class-give-tools-delete-test-transactions.php [hunk 1]
```diff
@@ -65,61 +63,10 @@ public function get_data() {
...
-
-						$sql[] = "DELETE FROM $wpdb->posts WHERE id IN ($ids)";
-						$sql[] = "DELETE FROM $wpdb->postmeta WHERE post_id IN ($ids)";
-						$sql[] = "DELETE FROM $wpdb->comments WHERE comment_post_ID IN ($ids)";
-						$sql[] = "DELETE FROM $wpdb->commentmeta WHERE comment_id NOT IN (SELECT comment_ID FROM $wpdb->comments)";
-						break;
-				}
-
+				// Delete the main payment.
+				give_delete_donation( (int) $item['id'] );
```
Comment (DevinWalker) [inline includes/admin/tools/data/class-give-tools-delete-test-transactions.php]: There's no fear of this breaking down with large site with lots of test donations? I can test with our GiveWP site. Have you tested @raftaar1191 ?

Comment (raftaar1191) [reply includes/admin/tools/data/class-give-tools-delete-test-transactions.php]: Yes I have Tested It Completely @DevinWalker

Comment (ravinderk) [reply includes/admin/tools/data/class-give-tools-delete-test-transactions.php]: @raftaar1191 Please use [`absint`](https://codex.wordpress.org/Function_Reference/absint)

Comment (raftaar1191) [reply includes/admin/tools/data/class-give-tools-delete-test-transactions.php]: Done Thanks @ravinderk to pointing this out

**Comment by raftaar1191** [comment]
Note: Why I did that 


`give_delete_donation` of Give Core functions handle all the related thing that we need to do 
1>Like deleting a log
2> Updating User
3> Deleting MetaData for the Donations
4> Delete Comment Related to Donations 
5> Recalculating Donations Form total count
5> Plus have some predefine hooks to handle the thing as well 

Here we can see that we are directly firing the SQL which is not good as we already have a specific function to delete the Donations. So Using it is an good options

**Comment by DevinWalker** [comment]
@raftaar1191 looks good. Merging.

**Pull request merged**